remotes/docker/authorizer.go

```diff
@@ -109,15 +121,27 @@ func (a *dockerAuthorizer) Authorize(ctx context.Context, req *http.Request) err
 		return nil
 	}
 
-	auth, err := ah.authorize(ctx)
+	auth, refreshToken, err := ah.authorize(ctx)
 	if err != nil {
 		return err
 	}
 
 	req.Header.Set("Authorization", auth)
+	a.mu.Lock()
+	a.fetchedRefreshToken = refreshToken
+	a.mu.Unlock()
```

Comment:
```suggestion
	if a.fetchedRefreshToken != "" {
		a.mu.Lock()
		a.fetchedRefreshToken = refreshToken
		a.mu.Unlock()
	}
```
when doing doBasicAuth, refreshToken may be empty which is no necessary for lock and update

Reply:
fixed